commons-vfs2-sandbox/src/main/java/org/apache/commons/vfs2/provider/smb2/Smb2ClientWrapper.java

```diff
@@ -0,0 +1,223 @@
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.List;
+
+/**
+ * A wrapper to the SMBClient for bundling the related client & connection instances.
+ */
+public class Smb2ClientWrapper extends SMBClient {
+
+    private static final SmbConfig CONFIG = SmbConfig.builder()
+            .withDfsEnabled(true)
+            .withMultiProtocolNegotiate(true)
+            .build();
+
+    protected final FileSystemOptions fileSystemOptions;
+    private final GenericFileName root;
+    private SMBClient smbClient;
+    private Connection connection;
+    private Session session;
+    private DiskShare diskShare;
+
+    protected Smb2ClientWrapper(final GenericFileName root, final FileSystemOptions fileSystemOptions)
+            throws FileSystemException {
+
+        this.root = root;
+        this.fileSystemOptions = fileSystemOptions;
+        smbClient = new SMBClient(CONFIG);
+        setupClient();
+    }
+
+    private void setupClient() throws FileSystemException {
+
+        final GenericFileName rootName = getRoot();
+
+        //the relevant data to authenticate a connection
+        String userName;
+        if (rootName.getUserName().equals("") || rootName.getUserName() == null) {
+            userName = "";
+        } else {
+            if (rootName.getUserName().contains(";")) {
+                userName = rootName.getUserName().substring(rootName.getUserName().indexOf(";") + 1, rootName.getUserName().length());
+            } else {
+                userName = rootName.getUserName();
+            }
+        }
+        String password = rootName.getPassword();
+        String authDomain = (rootName.getUserName().contains(";") ?
+                rootName.getUserName().substring(0, rootName.getUserName().indexOf(";")) : null);
+
+        //if username == "" the client tries to authenticate "anonymously". It's also possible to submit "guest" as username
+        AuthenticationContext authContext = new AuthenticationContext(userName, password.toCharArray(), authDomain);
+
+        //a connection stack is: SMBClient > Connection > Session > DiskShare
+        try {
+            connection = smbClient.connect(rootName.getHostName());
+            session = connection.authenticate(authContext);
+            String share = ((Smb2FileName) rootName).getShareName();
+            diskShare = (DiskShare) session.connectShare(share);
+        } catch (Exception e) {
+            throw new FileSystemException("vfs.provider.smb2/connect.error", rootName.getHostName(), e.getCause());
+        }
+    }
+
+    public GenericFileName getRoot() {
+
+        return root;
+    }
+
+    /**
+     * Returns file information of a given file.
+     *
+     * @param relativePath of file
+     * @return FileAllInformation object
+     */
+    public FileAllInformation getFileInfo(String relativePath) {
+
+        try {
+            return diskShare.getFileInformation(relativePath);
+        } catch (Exception e) {
+            return null;
```

Comment:
don't we need to log anything here.

Reply:
no. this will create error logs redundantly for lock files and when a file doesn't exist

Reply:
lets print debug logs

Reply:
done